skywalking-sniffer/skywalking-sdk-plugin/mongodb-3.x-plugin/src/main/java/com/a/eye/skywalking/plugin/mongodb/v3/MongoDBMethodInterceptor.java

```diff
@@ -0,0 +1,184 @@
+import com.a.eye.skywalking.api.plugin.interceptor.enhance.MethodInterceptResult;
+import com.a.eye.skywalking.trace.Span;
+import com.a.eye.skywalking.trace.tag.Tags;
+import com.mongodb.ReadPreference;
+import com.mongodb.bulk.DeleteRequest;
+import com.mongodb.bulk.InsertRequest;
+import com.mongodb.bulk.UpdateRequest;
+import com.mongodb.bulk.WriteRequest;
+import com.mongodb.operation.CountOperation;
+import com.mongodb.operation.CreateCollectionOperation;
+import com.mongodb.operation.CreateIndexesOperation;
+import com.mongodb.operation.CreateViewOperation;
+import com.mongodb.operation.DeleteOperation;
+import com.mongodb.operation.DistinctOperation;
+import com.mongodb.operation.FindAndDeleteOperation;
+import com.mongodb.operation.FindAndReplaceOperation;
+import com.mongodb.operation.FindAndUpdateOperation;
+import com.mongodb.operation.FindOperation;
+import com.mongodb.operation.GroupOperation;
+import com.mongodb.operation.InsertOperation;
+import com.mongodb.operation.ListCollectionsOperation;
+import com.mongodb.operation.MapReduceToCollectionOperation;
+import com.mongodb.operation.MapReduceWithInlineResultsOperation;
+import com.mongodb.operation.MixedBulkWriteOperation;
+import com.mongodb.operation.ReadOperation;
+import com.mongodb.operation.UpdateOperation;
+import com.mongodb.operation.WriteOperation;
+
+/**
+ * {@link MongoDBMethodInterceptor} intercept method of {@link com.mongodb.Mongo#execute(ReadOperation, ReadPreference)}
+ * or {@link com.mongodb.Mongo#execute(WriteOperation)}. record the mongoDB host, operation name and the key of the
+ * operation.
+ *
+ * @author baiyang
+ */
+public class MongoDBMethodInterceptor implements InstanceMethodsAroundInterceptor {
+
+    /**
+     * The key name that MongoDB host in {@link EnhancedClassInstanceContext#context}.
+     */
+    static final String MONGODB_HOST = "MONGODB_HOST";
+
+    /**
+     * The key name that MongoDB port in {@link EnhancedClassInstanceContext#context}.
+     */
+    static final String MONGODB_PORT = "MONGODB_PORT";
+
+    private static final String MONGODB_COMPONENT = "MongoDB";
+
+    private static final String METHOD = "MongoDB/";
+
+    private static final int FILTER_LENGTH_LIMIT = 256;
+
+    @Override
+    public void beforeMethod(final EnhancedClassInstanceContext context,
+            final InstanceMethodInvokeContext interceptorContext, final MethodInterceptResult result) {
+        Object[] arguments = interceptorContext.allArguments();
+        Span span = null;
+        if (Config.Agent.MONGODB_BINDPARAM) {
+            OperationInfo operationInfo = this.getReadOperationInfo(arguments[0]);
+            span = ContextManager.createSpan(METHOD + operationInfo.getMethodName());
+            Tags.DB_STATEMENT.set(span, operationInfo.getMethodName() + " " + operationInfo.getFilter());
+        } else {
+            span = ContextManager.createSpan(METHOD + arguments[0].getClass().getSimpleName());
```

Comment:
By `Config.Agent.MONGODB_BINDPARAM`, you set `OperationName` in two different circumstances, and you set it different. This is unacceptable to me.

The `OperationName` is the most important for a span, it should be changed just because you decided to trace parameters.